torch/csrc/jit/passes/convert_activation.cpp

```diff
@@ -0,0 +1,100 @@
+#include <torch/csrc/jit/passes/convert_activation.h>
+#include <torch/csrc/jit/passes/remove_mutation.h>
+#include "ATen/core/interned_strings.h"
+
+namespace torch {
+namespace jit {
+
+namespace {
+static const std::unordered_set<Symbol> inplace_activation_ops = {
+    aten::hardsigmoid_,
+    aten::hardtanh_,
+    aten::hardswish_,
+    aten::relu_,
+    aten::relu6_,
+    aten::sigmoid_,
+    aten::tanh_};
+} // namespace
+
+FunctionalActivationRewriter::FunctionalActivationRewriter(
+    std::shared_ptr<Graph> graph)
+    : aliasDb_(nullptr), graph_(std::move(graph)) {
+  for (auto op : inplace_activation_ops) {
+    std::string name = std::string(op.toQualString());
+    name.pop_back();
+    functional_activation_ops_.insert(Symbol::fromQualString(name));
+  }
+}
+
+bool FunctionalActivationRewriter::FunctionalToInplaceActivation(Block* block) {
+  bool changed = false;
+  for (auto it = block->nodes().begin(); it != block->nodes().end();) {
+    auto* node = *it;
+    it++;
+
+    for (Block* sub_block : node->blocks()) {
+      changed |= FunctionalToInplaceActivation(sub_block);
+    }
+
+    if (functional_activation_ops_.count(node->kind()) == 0) {
+      continue;
+    }
+
+    Value* input = node->inputs().at(0);
+    Value* output = node->output();
+
+    // Use y = relu(x) as an example:
```

Comment:
The key part of the algorithm is to decide which activation functions can be rewritten to the in-place form. Could you carve out that part of the logic into a separate helper function and describe the algorithm in the comments? I feel that the current criteria is safe but not optimal and may be improved later and labeling this part of the algorithm separately may make it easier for future update